tests/test_read_into_dict.py

```diff
@@ -0,0 +1,41 @@
+"""Tests for the csvy reader functions."""
+
+import logging
+import warnings
+
+import pytest
+
+from csvy.readers import read_into_dict
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+data_files = [
+    "tests/data_comment.csv",
+    "tests/array_data.csv",
+    "tests/data.csv",
+    "tests/data_read_into_dict.csvy",
+]
+
+
+@pytest.mark.parametrize("file_path", data_files)
+def test_read_into_dict(file_path):
```

Comment:
This test function should be moved to `test_read.py`, which is where all the test files for the readers are.

Also, it should use the existing fixtures for the data file paths. The `data_path` fixture alone is probably satisfactory for testing the function.